File: documentation/user/en/get-started/example/create-small-dataset.cs
evita.UpdateCatalog(
	"evita",
	session => {
		session
			.CreateNewEntity("Brand", 2)
			.SetAttribute("name", "Lenovo")
			.UpsertVia(session);

		session
			.CreateNewEntity("Brand", 3)
			.SetAttribute("name", "Acer")
			.UpsertVia(session);

		session
			.CreateNewEntity("Brand", 4)
			.SetAttribute("name", "ASUS")
			.UpsertVia(session);

		session
			.CreateNewEntity("Category", 1)
			.SetAttribute("name", new CultureInfo("en"), "Electronics")
			.SetAttribute("name", new CultureInfo("de"), "Elektronik")
			.UpsertVia(session);

		session
			.CreateNewEntity("Category", 2)
			.SetParent(1)
			.SetAttribute("name", new CultureInfo("en"), "Components")
			.SetAttribute("name", new CultureInfo("de"), "Komponenten")
			.UpsertVia(session);

		session
			.CreateNewEntity("Category", 3)
			.SetParent(1)
			.SetAttribute("name", new CultureInfo("en"), "Portable computer")
			.SetAttribute("name", new CultureInfo("de"), "Tragbarer Computer")
			.UpsertVia(session);

		session.CreateNewEntity("Product")
			.SetAttribute("name", new CultureInfo("en"), "Lenovo ThinkPad UltraSlim USB DVD Burner")
			.SetAttribute("name", new CultureInfo("de"), "Lenovo ThinkPad UltraSlim USB-DVD-Brenner")
			.SetAttribute("catalogCode", "4XA0E97775")
			.SetAttribute("stockQuantity", 3)
			.SetAssociatedData(
				"gallery",
				new string[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NT442p2i&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NT442p2i-02&i=1.jpg"
				}
			)
			.SetPrice(
				1, "basic", new Currency("EUR"),
				63.93m, 22m, 78m,
				true
			)
			.SetPrice(
				2, "basic", new Currency("USD"),
				68.03m, 22m, 83m,
				true
			)
			.SetReference("brand", "Brand", Cardinality.ZeroOrMore, 2)
			.SetReference("categories", "Category", Cardinality.ZeroOrMore, 2)
			.UpsertVia(session);

		session.CreateNewEntity("Product")
			.SetAttribute("name", new CultureInfo("en"), "ASUS SDRW-08U7M-U black + 2× M-Disk")
			.SetAttribute("name", new CultureInfo("de"), "ASUS SDRW-08U7M-U schwarz + 2× M-Disk")
			.SetAttribute("catalogCode", "90DD01X0-M29000")
			.SetAttribute("stockQuantity", 1)
			.SetAssociatedData(
				"gallery",
				new string[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=GM382c8d&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=GM382c8d-02&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=GM382c8d-03&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=GM382c8d-04&i=1.jpg"
				}
			)
			.SetPrice(
				1, "basic", new Currency("EUR"),
				27.87m, 22m, 34m,
				true
			)
			.SetPrice(
				2, "basic", new Currency("USD"),
				29.5m, 22m, 36m,
				true
			)
			.SetReference("brand", "Brand", Cardinality.ZeroOrMore, 4)
			.SetReference("categories", "Category", Cardinality.ZeroOrMore, 2)
			.UpsertVia(session);

		session.CreateNewEntity("Product")
			.SetAttribute("name", new CultureInfo("en"), "Lenovo Legion 5 15ITH6H Phantom Blue/Shadow Black(3 years warranty)")
			.SetAttribute("name", new CultureInfo("de"), "Lenovo Legion 5 15ITH6H Phantom Blau/Schatten Schwarz(3 Jahre Garantie)")
			.SetAttribute("catalogCode", "82JH00KYCK")
			.SetAttribute("stockQuantity", 8)
			.SetAssociatedData(
				"gallery",
				new string[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NT379t71j3b&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NT379t71j3b-01&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NT379t71j3b-06&i=1.jpg"
				}
			)
			.SetPrice(
				1, "basic", new Currency("EUR"),
				1040.16m, 22m, 1269m,
				true
			)
			.SetPrice(
				2, "basic", new Currency("USD"),
				1097.54m, 22m, 1339m,
				true
			)
			.SetReference("brand", "Brand", Cardinality.ZeroOrMore, 2)
			.SetReference("categories", "Category", Cardinality.ZeroOrMore, 3)
			.UpsertVia(session);

		session.CreateNewEntity("Product")
			.SetAttribute("name", new CultureInfo("en"), "Acer Nitro 5 Shale Black")
			.SetAttribute("name", new CultureInfo("de"), "Acer Nitro 5 Shale Schwarz")
			.SetAttribute("catalogCode", "NH.QEKEC.002")
			.SetAttribute("stockQuantity", 6)
			.SetAssociatedData(
				"gallery",
				new string[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NC108c7i05a8b&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NC108c7i05a8b-01&i=1.jpg"
				}
			)
			.SetPrice(
				1, "basic", new Currency("EUR"),
				627.05m, 22m, 765m,
				true
			)
			.SetPrice(
				2, "basic", new Currency("USD"),
				654.92m, 22m, 799m,
				true
			)
			.SetReference("brand", "Brand", Cardinality.ZeroOrMore, 3)
			.SetReference("categories", "Category", Cardinality.ZeroOrMore, 3)
			.UpsertVia(session);

		session.CreateNewEntity("Product")
			.SetAttribute("name", new CultureInfo("en"), "ASUS Vivobook 16 X1605EA-MB044W Indie Black")
			.SetAttribute("name", new CultureInfo("de"), "ASUS Vivobook 16 X1605EA-MB044W Indie Schwarz")
			.SetAttribute("catalogCode", "X1605EA-MB044W")
			.SetAttribute("stockQuantity", 1)
			.SetAssociatedData(
				"gallery",
				new string[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NA579p8e0&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NA579p8e0-04&i=1.jpg"
				}
			)
			.SetPrice(
				1, "basic", new Currency("EUR"),
				345.9m, 22m, 422m,
				true
			)
			.SetPrice(
				2, "basic", new Currency("USD"),
				365.57m, 22m, 446m,
				true
			)
			.SetReference("brand", "Brand", Cardinality.ZeroOrMore, 4)
			.SetReference("categories", "Category", Cardinality.ZeroOrMore, 3)
			.UpsertVia(session);
	}
);

File: documentation/user/en/get-started/example/create-small-dataset.java
/*
 *
 *                         _ _        ____  ____
 *               _____   _(_) |_ __ _|  _ \| __ )
 *              / _ \ \ / / | __/ _` | | | |  _ \
 *             |  __/\ V /| | || (_| | |_| | |_) |
 *              \___| \_/ |_|\__\__,_|____/|____/
 *
 *   Copyright (c) 2024
 *
 *   Licensed under the Business Source License, Version 1.1 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *   https://github.com/FgForrest/evitaDB/blob/master/LICENSE
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 */

evita.updateCatalog(
	"evita",
	session -> {
		session
			.createNewEntity("Brand", 2)
			.setAttribute("name", "Lenovo")
			.upsertVia(session);

		session
			.createNewEntity("Brand", 3)
			.setAttribute("name", "Acer")
			.upsertVia(session);

		session
			.createNewEntity("Brand", 4)
			.setAttribute("name", "ASUS")
			.upsertVia(session);

		session
			.createNewEntity("Category", 1)
			.setAttribute("name", Locale.ENGLISH, "Electronics")
			.setAttribute("name", Locale.GERMAN, "Elektronik")
			.upsertVia(session);

		session
			.createNewEntity("Category", 2)
			.setParent(1)
			.setAttribute("name", Locale.ENGLISH, "Components")
			.setAttribute("name", Locale.GERMAN, "Komponenten")
			.upsertVia(session);

		session
			.createNewEntity("Category", 3)
			.setParent(1)
			.setAttribute("name", Locale.ENGLISH, "Portable computer")
			.setAttribute("name", Locale.GERMAN, "Tragbarer Computer")
			.upsertVia(session);

		session.createNewEntity("Product")
			.setAttribute("name", Locale.ENGLISH, "Lenovo ThinkPad UltraSlim USB DVD Burner")
			.setAttribute("name", Locale.GERMAN, "Lenovo ThinkPad UltraSlim USB-DVD-Brenner")
			.setAttribute("catalogCode", "4XA0E97775")
			.setAttribute("stockQuantity", 3)
			.setAssociatedData(
				"gallery",
				new String[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NT442p2i&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NT442p2i-02&i=1.jpg"
				}
			)
			.setPrice(
				1, "basic", Currency.getInstance("EUR"),
				new BigDecimal("63.93"), new BigDecimal("22"), new BigDecimal(78),
				true
			)
			.setPrice(
				2, "basic", Currency.getInstance("USD"),
				new BigDecimal("68.03"), new BigDecimal("22"), new BigDecimal(83),
				true
			)
			.setReference("brand", "Brand", Cardinality.ZERO_OR_ONE, 2)
			.setReference("categories", "Category", Cardinality.ZERO_OR_MORE, 2)
			.upsertVia(session);

		session.createNewEntity("Product")
			.setAttribute("name", Locale.ENGLISH, "ASUS SDRW-08U7M-U black + 2× M-Disk")
			.setAttribute("name", Locale.GERMAN, "ASUS SDRW-08U7M-U schwarz + 2× M-Disk")
			.setAttribute("catalogCode", "90DD01X0-M29000")
			.setAttribute("stockQuantity", 1)
			.setAssociatedData(
				"gallery",
				new String[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=GM382c8d&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=GM382c8d-02&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=GM382c8d-03&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=GM382c8d-04&i=1.jpg"
				}
			)
			.setPrice(
				1, "basic", Currency.getInstance("EUR"),
				new BigDecimal("27.87"), new BigDecimal("22"), new BigDecimal(34),
				true
			)
			.setPrice(
				2, "basic", Currency.getInstance("USD"),
				new BigDecimal("29.5"), new BigDecimal("22"), new BigDecimal(36),
				true
			)
			.setReference("brand", "Brand", Cardinality.ZERO_OR_ONE, 4)
			.setReference("categories", "Category", Cardinality.ZERO_OR_MORE, 2)
			.upsertVia(session);

		session.createNewEntity("Product")
			.setAttribute("name", Locale.ENGLISH, "Lenovo Legion 5 15ITH6H Phantom Blue/Shadow Black(3 years warranty)")
			.setAttribute("name", Locale.GERMAN, "Lenovo Legion 5 15ITH6H Phantom Blau/Schatten Schwarz(3 Jahre Garantie)")
			.setAttribute("catalogCode", "82JH00KYCK")
			.setAttribute("stockQuantity", 8)
			.setAssociatedData(
				"gallery",
				new String[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NT379t71j3b&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NT379t71j3b-01&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NT379t71j3b-06&i=1.jpg"
				}
			)
			.setPrice(
				1, "basic", Currency.getInstance("EUR"),
				new BigDecimal("1040.16"), new BigDecimal("22"), new BigDecimal(1269),
				true
			)
			.setPrice(
				2, "basic", Currency.getInstance("USD"),
				new BigDecimal("1097.54"), new BigDecimal("22"), new BigDecimal(1339),
				true
			)
			.setReference("brand", "Brand", Cardinality.ZERO_OR_ONE, 2)
			.setReference("categories", "Category", Cardinality.ZERO_OR_MORE, 3)
			.upsertVia(session);

		session.createNewEntity("Product")
			.setAttribute("name", Locale.ENGLISH, "Acer Nitro 5 Shale Black")
			.setAttribute("name", Locale.GERMAN, "Acer Nitro 5 Shale Schwarz")
			.setAttribute("catalogCode", "NH.QEKEC.002")
			.setAttribute("stockQuantity", 6)
			.setAssociatedData(
				"gallery",
				new String[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NC108c7i05a8b&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NC108c7i05a8b-01&i=1.jpg"
				}
			)
			.setPrice(
				1, "basic", Currency.getInstance("EUR"),
				new BigDecimal("627.05"), new BigDecimal("22"), new BigDecimal(765),
				true
			)
			.setPrice(
				2, "basic", Currency.getInstance("USD"),
				new BigDecimal("654.92"), new BigDecimal("22"), new BigDecimal(799),
				true
			)
			.setReference("brand", "Brand", Cardinality.ZERO_OR_ONE, 3)
			.setReference("categories", "Category", Cardinality.ZERO_OR_MORE, 3)
			.upsertVia(session);

		session.createNewEntity("Product")
			.setAttribute("name", Locale.ENGLISH, "ASUS Vivobook 16 X1605EA-MB044W Indie Black")
			.setAttribute("name", Locale.GERMAN, "ASUS Vivobook 16 X1605EA-MB044W Indie Schwarz")
			.setAttribute("catalogCode", "X1605EA-MB044W")
			.setAttribute("stockQuantity", 1)
			.setAssociatedData(
				"gallery",
				new String[] {
					"https://cdn.alza.cz/ImgW.ashx?fd=f4&cd=NA579p8e0&i=1.jpg",
					"https://cdn.alza.cz/ImgW.ashx?fd=FotoAddOrig&cd=NA579p8e0-04&i=1.jpg"
				}
			)
			.setPrice(
				1, "basic", Currency.getInstance("EUR"),
				new BigDecimal("345.9"), new BigDecimal("22"), new BigDecimal(422),
				true
			)
			.setPrice(
				2, "basic", Currency.getInstance("USD"),
				new BigDecimal("365.57"), new BigDecimal("22"), new BigDecimal(446),
				true
			)
			.setReference("brand", "Brand", Cardinality.ZERO_OR_ONE, 4)
			.setReference("categories", "Category", Cardinality.ZERO_OR_MORE, 3)
			.upsertVia(session);
	}
);
